ethers-middleware/src/proxy_wallet/ds_proxy/mod.rs

```diff
@@ -0,0 +1,89 @@
+use super::{ProxyWallet, ProxyWalletError};
+use ethers_contract::BaseContract;
+use ethers_core::{
+    abi::{parse_abi, Tokenize},
+    types::*,
+};
+
+const DS_PROXY_EXECUTE_SIGNATURE: &str =
+    "function execute(address target, bytes data) public payable returns (bytes memory response)";
+
+#[derive(Debug)]
+pub struct DsProxy {
+    address: Address,
+}
+
+impl DsProxy {
+    /// Create a new instance of DsProxy by providing the address of the DsProxy contract that has
+    /// already been deployed to the Ethereum network.
+    pub fn new(address: Address) -> Self {
+        Self { address }
+    }
+}
```

Comment:
Should we also add an extra method which calls `function execute(bytes _code, bytes _data)`?